drud/ddev
# Use caching filesystem to improve webserving performance

## The Problem/Issue/Bug:

ddev webserving behavior is slower than we'd like on macOS and Windows because we're serving from a docker bind-mount, and on those platforms it's not as good as it ought to be. Linux is fine.

## How this PR Solves The Problem:

Use a fork of cweagans/bg-sync to sync between the bind-mounted filesystem and a much faster docker native volume. bgsync uses a container with unison running on it.

This is experimental and currently tested only on macOS, as there are filesystem problems on Windows. 

## Manual Testing Instructions:

* Make sure you have the latest version of the bgsync container with `docker pull drud/ddev-bgsync:20181117_bgsync`
* In .ddev/config.yaml set `webcache_enabled: true`
* `ddev start`. Do things. Verify behavior. See how fast it is. Try to figure out how to make it unreliable.

## Automated Testing Overview:

* For macOS we run the entire test suite.
* A number of tests had to be adapted to wait for sync before continuing.
* I would like to see standalone tests for ddev-bgsync container, based on BATS probably.

## Related Issue Link(s):

## Release/Deployment notes:
